### PR TITLE
IBM FHIR Server audit does not log bundles correctly. #1803

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java [hunk 1]
```diff
@@ -82,7 +82,7 @@
     private static CadfResource observerRsrc =
             new CadfResource.Builder("fhir-server", ResourceType.compute_node)
                             .geolocation(new CadfGeolocation.Builder(geoCity, geoState, geoCountry, null).build())
-                            .name("Fhir Audit")
+                            .name("IBM FHIR Server - Audit")
                             .host(System.getenv("HOSTNAME"))
```
Comment (punktilious) [inline fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java:86]: I know it's not part of the change, but is this the best way to get the name of the host? Do we ensure that HOSTNAME is part of the environment?  On Windows, COMPUTERNAME might be more appropriate. Let me know if you think we need a ticket for this.

File: fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java [hunk 2]
```diff
@@ -197,8 +196,10 @@ public static CadfEvent createCadfEvent(AuditLogEntry logEntry)
         Outcome cadfEventOutCome;
 
         // Cadf does't log config, so skip
-        if ((logEntry.getEventType() != AuditLogEventType.FHIR_CONFIGDATA.value()) && logEntry.getContext() != null
-                && logEntry.getContext().getAction() != null && logEntry.getContext().getApiParameters() != null) {
+        if ((logEntry.getEventType() != AuditLogEventType.FHIR_CONFIGDATA.value())
```
Comment (punktilious) [inline fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java:199]: String != String. Shouldn't this use equals(...)?

File: fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java (hunk 1 above)
Comment (prb112) [reply fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java:86]: Good point, I don't think it's super reliable. I think a ticket isn't bad, probably lower priority.

File: fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java (hunk 2 above)
Comment (prb112) [reply fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java]: Cleaned this up by addressing Lee's well-known request

File: fhir-server/src/main/java/com/ibm/fhir/server/util/RestAuditLogger.java [hunk 3]
```diff
@@ -311,6 +313,21 @@ public static void logBundle(HttpServletRequest request, Bundle bundle, Date sta
                 .resourcesUpdated(updateCount).build());
         entry.setDescription("FHIR Bundle request");
 
+        // Previously we didn't set the Action which caused the logEntry to
+        // Skip over the actual logging, int his case, we're deciding by default
```
Comment (lmsurpre) [inline fhir-server/src/main/java/com/ibm/fhir/server/util/RestAuditLogger.java:317]:
```suggestion
        // Skip over the actual logging, in this case, we're deciding by default
```

**Review by lmsurpre — APPROVED** [review]
LGTM

File: fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java (hunk 1 above)
Comment (prb112) [reply fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java:100]: Thanks

File: fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java]: outdated now, but AuditLogEventType is an enum and enum==enum is just fine.

**Pull request merged**